src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/UploadEnvelopeDocumentsService.java

```diff
@@ -0,0 +1,233 @@
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.document.output.Pdf;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.BlobManager;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.DocumentProcessor;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.EnvelopeProcessor;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.ZipFileProcessingResult;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.ZipFileProcessor;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.zip.ZipInputStream;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.DOC_UPLOADED;
+import static uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event.DOC_UPLOAD_FAILURE;
+
+/**
+ * Service responsible to upload envelopes ended in state after main processor task.
+ * <p></p>
+ * {@link uk.gov.hmcts.reform.bulkscanprocessor.tasks.BlobProcessorTask}
+ * <p></p>
+ * All the validations have been accomplished and {@link Envelope} saved in DB.
+ * Therefore no need to go through the same set of validation steps/rules and just re-extract zip file and upload.
+ * In case some {@link StorageException} or {@link IOException} experienced during zip extraction
+ * the envelope state will not be changed and left for a retry on the next run.
+ * {@link Event#DOC_UPLOAD_FAILURE} can be treated as stuck envelope and be re-uploaded by same service here.
+ * There is an upload retry counter/limit which may be used as well
+ * Review ^ last sentence and, perhaps, incorporate such possibility here.
+ */
+@Service
+public class UploadEnvelopeDocumentsService {
+
+    private static final Logger log = getLogger(UploadEnvelopeDocumentsService.class);
+
+    private final BlobManager blobManager;
+    private final ZipFileProcessor zipFileProcessor;
+    private final DocumentProcessor documentProcessor;
+    private final EnvelopeProcessor envelopeProcessor;
+
+    public UploadEnvelopeDocumentsService(
+        BlobManager blobManager,
+        ZipFileProcessor zipFileProcessor,
+        DocumentProcessor documentProcessor,
+        EnvelopeProcessor envelopeProcessor
+    ) {
+        this.blobManager = blobManager;
+        this.zipFileProcessor = zipFileProcessor;
+        this.documentProcessor = documentProcessor;
+        this.envelopeProcessor = envelopeProcessor;
+    }
+
+    public void processByContainer(String containerName, List<Envelope> envelopes) {
+        log.info("Processing envelopes in {} container. Envelopes found: {}", containerName, envelopes.size());
```

Comment:
I wouldn't say those envelopes were _found_ here.
Perhaps and old log from when this service was also fetching envelopes from DB?